### PR TITLE
feat: entity query selections to projections in doc store

### PR DESCRIPTION
**Pull request by skjindal93**
## Description
Flowing down the selections requested on EntityQueryRequest to doc store projections. Depends on https://github.com/hypertrace/document-store/pull/28

### Testing
Added unit tests. Verified end to end by port forwarding various services and loading all the screens on the UI

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: entity-service-impl/build.gradle.kts [hunk 1]
```diff
@@ -7,7 +7,7 @@ plugins {
 dependencies {
   api(project(":entity-service-api"))
   api("org.hypertrace.core.serviceframework:service-framework-spi:0.1.19")
-  implementation("org.hypertrace.core.documentstore:document-store:0.4.6")
+  implementation("org.hypertrace.core.documentstore:document-store:0.4.9-SNAPSHOT")
```
Comment (skjindal93) [inline entity-service-impl/build.gradle.kts:10]: Will change once https://github.com/hypertrace/document-store/pull/28 is merged

File: entity-service/build.gradle.kts [hunk 2]
```diff
@@ -59,7 +59,7 @@ dependencies {
   implementation("org.hypertrace.core.grpcutils:grpc-server-utils:0.3.1")
   implementation("org.hypertrace.core.grpcutils:grpc-client-utils:0.3.1")
   implementation("org.hypertrace.core.serviceframework:platform-service-framework:0.1.19")
-  implementation("org.hypertrace.core.documentstore:document-store:0.4.6")
+  implementation("org.hypertrace.core.documentstore:document-store:0.4.9-SNAPSHOT")
```
Comment (skjindal93) [inline entity-service/build.gradle.kts:62]: Will change once https://github.com/hypertrace/document-store/pull/28 is merged

File: entity-service-impl/src/test/java/org/hypertrace/entity/query/service/EntityQueryConverterTest.java [hunk 3]
```diff
@@ -114,4 +115,23 @@ public void test_convertToEdsQuery_functionExpressionOrderByExpression_throwsExc
     assertThrows(UnsupportedOperationException.class,
         () -> EntityQueryConverter.convertToEDSQuery(request, Collections.emptyMap()));
   }
+
```
Comment (avinashkolluru) [inline entity-service-impl/src/test/java/org/hypertrace/entity/query/service/EntityQueryConverterTest.java:118]: Are there no IntegrationTests for EntityQueryService?
I think we should add one.

Comment (skjindal93) [reply entity-service-impl/src/test/java/org/hypertrace/entity/query/service/EntityQueryConverterTest.java:118]: There weren't any. Let me try to add them

Comment (skjindal93) [reply entity-service-impl/src/test/java/org/hypertrace/entity/query/service/EntityQueryConverterTest.java:118]: added integration tests

**Review by avinashkolluru — APPROVED** [review]

**Pull request merged**